Crashlytics/UnitTests/FIRRecordExceptionModelTests.m

```diff
@@ -14,7 +14,8 @@
 
 #import <XCTest/XCTest.h>
 
-#import "Crashlytics/Crashlytics/Public/FirebaseCrashlytics/FIRExceptionModel.h"
+#import <FirebaseCrashlytics/FIRExceptionModel.h>
+//#import "Crashlytics/Crashlytics/Public/FirebaseCrashlytics/FIRExceptionModel.h"
```

Comment:
When changing the imports from "..." → <...> in the public headers, this surfaced an issue when building the target's test specs (causing most/all of the pod-lib-lint failures on CI).

I also had to change imports _of_ the public headers to the bracket notation to avoid the redefinition errors:
<img width="924" alt="Screenshot 2023-11-07 at 9 23 37 AM" src="https://github.com/firebase/firebase-ios-sdk/assets/36927374/28c879d7-9941-4921-aee7-d353d0c1b2d1">

Reply:
I think the issue has to do with the public headers not forward declaring public types where they could be. It would be a breaking change to change that right now though.

Reply:
@paulb777, I'm not too sure about this. Do you see anything troubling about changing all non-public imports of public headers from `"..."` → `<...>`?